binding/ios/Orca.swift

```diff
@@ -0,0 +1,359 @@
+            modelPathArg = try getResourcePath(modelPathArg!)
+        }
+
+        pv_set_sdk(Orca.sdk)
+
+        let status = pv_orca_init(accessKey, modelPathArg, &handle)
+        if status != PV_STATUS_SUCCESS {
+            let messageStack = try getMessageStack()
+            throw pvStatusToOrcaError(status, "Orca init failed", messageStack)
+        }
+    }
+
+    /// Constructor.
+    ///
+    /// - Parameters:
+    ///   - accessKey: The AccessKey obtained from Picovoice Console (https://console.picovoice.ai).
+    ///   - modelURL: URL to file containing model parameters.
+    /// - Throws: OrcaError
+    public convenience init(
+            accessKey: String,
+            modelURL: URL) throws {
+
+        try self.init(
+                accessKey: accessKey,
+                modelPath: modelURL.path)
+    }
+
+    deinit {
+        self.delete()
+    }
+
+    /// Releases native resources that were allocated to Orca
+    public func delete() {
+        if handle != nil {
+            pv_orca_delete(handle)
+            handle = nil
+        }
+    }
+
+    /// Generates audio from text. The returned audio contains the speech representation of the text.
+    ///
+    /// - Parameters:
+    ///   - text: Text to be converted to audio. The maximum number of characters per call to `.synthesize()` is
+    ///    `self.maxCharacterLimit`. Allowed characters can be retrieved by calling `self.validCharacters`.
+    ///    Custom pronunciations can be embedded in the text via the syntax `{word|pronunciation}`.
+    ///    The pronunciation is expressed in ARPAbet format, e.g.: "I {live|L IH V} in {Sevilla|S EH V IY Y AH}".
+    ///   - speechRate: Rate of speech of the generated audio.
+    /// - Returns: The generated audio, stored as a sequence of 16-bit linearly-encoded integers.
+    /// - Throws: OrcaError
+    public func synthesize(text: String, speechRate: Double? = nil) throws -> [Int16] {
+        if handle == nil {
+            throw OrcaInvalidStateError("Unable to synthesize - resources have been released")
+        }
+
+        if text.count > Orca.maxCharacterLimit {
+            throw OrcaInvalidArgumentError(
+                "Text length (\(text.count)) must be smaller than \(Orca.maxCharacterLimit)")
+        }
+
+        let characters = try self.validCharacters
+        let regex = try NSRegularExpression(
+            pattern: "[^\(characters.joined(separator: ""))\\s{}|']",
+            options: .caseInsensitive)
+        let range = NSRange(text.startIndex..<text.endIndex, in: text)
+        let matches = regex.matches(in: text, range: range)
+
+        let unexpectedCharacters = matches.map {
+            String(text[Range($0.range, in: text)!])
+        }
+
+        if unexpectedCharacters.count > 0 {
+            throw OrcaInvalidArgumentError(
+                "Text contains the following invalid characters: `\(unexpectedCharacters.joined(separator: ", "))`")
+        }
```

Comment:
I didn't do this validation in my SDK because it wasn't being done in the Python SDK - isn't this something already handled in PV_ERROR?

Reply:
also, if you enter many of the same invalid character, the exception will print all instances of it. (e.g. text = "444", "Text contains the following invalid characters: 4, 4, 4")